Apps/Gate/gate.c

```diff
@@ -90,10 +66,9 @@ void Gate_Loop(void)
             if ((Luos_GetSystick() - last_time >= TimeOD_TimeTo_ms(update_time)) && (Luos_GetSystick() > last_time))
             {
                 last_time = Luos_GetSystick();
-                DataManager_Run(gate);
-#ifndef GATE_POLLING
-                if (first_conversion == 1)
+                if (end_detection)
```

Comment:
You should put the configurations on the `if (data_msg->header.cmd == END_DETECTION)` of data_manager.c.